constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/Spans.java

```diff
@@ -64,6 +69,37 @@ public Spans filter(final Predicate<Window> filter) {
     return new Spans(this.windows.stream().filter(filter).toList());
   }
 
+  @Override
+  public Spans split(final int numberOfSubSpans) {
+    if (numberOfSubSpans == 1) {
+      return new Spans(this);
+    }
+    return this.flatMap(x -> {
+      // Width of each sub-window, rounded down to the microsecond
+      final var width = Duration.divide(Duration.subtract(x.end, x.start), numberOfSubSpans);
```

Comment:
I am a little confused by this. 

`x` - window
`x.end -x.start` - span/width of the window?

Why are you dividing by `numberOfSubSpans` are all the sub-window the same length?

Reply:
The variable name is a little misleading I guess, but it explains in the comment above. `width` here is the width of the new sub-windows, not the width of the original window. And no, if the number of microseconds in the original window is not a multiple of `N`, then the last sub-window can be up to `N-1` microseconds longer than the rest. So they're not *exactly* the same width, but they're close.

Reply:
And to be clear, this piece of code is for the `split` operation, which is a different PR. This PR is for `starts` and `ends`, only the last two commits. I should have put that in the main description instead of just a comment.